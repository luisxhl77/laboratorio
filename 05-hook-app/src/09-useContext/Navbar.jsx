import { Link, NavLink } from "react-router-dom"

export const Navbar = () => {
    return (
        <nav className="navbar navbar-expand-lg bg-dark navbar-dark">
            <div className="container-fluid">

                <Link className="navbar-brand" to="/">useContext</Link>

                <div className="collapse navbar-collapse" id="navbarNav">

                    <ul className="navbar-nav">
                    
                    <NavLink to ="/" className={(args) => {
                        return 'nav-link';
                    } }>
                        Home
                    </NavLink>

                    <NavLink to ="/login" className={(args) => {
                        return 'nav-link';
                    } }>
                        Login
                    </NavLink>

                    <NavLink to ="/about" className={(args) => {
                        return 'nav-link';
                    } }>
                        About
                    </NavLink>

                    </ul>

                </div>
            </div>
        </nav>
    )
}
