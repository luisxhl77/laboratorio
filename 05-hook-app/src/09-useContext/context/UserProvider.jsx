import { useState } from "react"
import { Usercontext } from "./userContext"

const user = {
    id: 123,
    name: 'fernando herrera',
    email: 'herrera@gmail.com',
}

export const UserProvider = ({ children }) => {

    const [user, setUser] = useState();

  return (
        <Usercontext.Provider value={{user, setUser}}>
            {children}
        </Usercontext.Provider>
    )
}
