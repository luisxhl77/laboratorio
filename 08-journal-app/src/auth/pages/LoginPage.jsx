import { Grid, Typography, TextField, Button} from '@mui/material'
import {Google} from '@mui/icons-material';
import { Link } from "react-router-dom";
import { AuthLayout } from '../layout/AuthLayout';
import { useForm } from '../../hooks';
import { checkingAuthentication, startGoogleSignIn } from '../../store/auth/thunks';
import { useDispatch } from 'react-redux';

export const LoginPage = () => {

  const dispatch = useDispatch();

  const {email, password, onInputChange, onResetForm,} = useForm({
    email: 'luisxhl7@gmail.com',
    password: 'luchocarlos123777'
  })

  const onSubmit = ( event ) => {
    event.preventDefault();
    console.log({ email, password });
  }

  const onGoogleLogin = () => {
    dispatch(checkingAuthentication());
  }
  const onGoogleSingnIn = () => {
    dispatch(startGoogleSignIn());
  }


  return (
    <AuthLayout title='Login'>
      <form onSubmit={ onSubmit }>
        <Grid item xs={12} sx={{ mt: 2 }}>
          <TextField label="correo" type="email" placeholder="example@gmail.com" fullWidth name="email"  value={email} onChange={ onInputChange }/>
        </Grid>

        <Grid item xs={12} sx={{ mt: 2 }}>
          <TextField label="contraseña" type="password" placeholder="contraseña" fullWidth name="password"  value={password} onChange={ onInputChange }/>
        </Grid>

        <Grid container spacing={ 2 } sx={{ mb: 2, mt: 1 }}>
          <Grid item xs={12} sm={6}>
            <Button type='submit' variant="contained" fullWidth>
              Login
            </Button>
          </Grid>
          <Grid item xs={12} sm={6}>
            <Button variant="contained" fullWidth onClick={ onGoogleSingnIn }>
              <Google/>
              <Typography sx={{ ml: 1}}> Google </Typography>
            </Button>
          </Grid>
        </Grid>

        <Grid container direction='row' justifyContent='end' >
          <Link color='inherit' to='/auth/register'>
            Crear una cuenta
          </Link>
        </Grid>
      </form>
    </AuthLayout>
  )
}
