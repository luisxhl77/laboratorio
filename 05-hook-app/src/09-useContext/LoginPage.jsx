import { useContext } from "react"
import { Usercontext } from "./context/userContext";

export const LoginPage = () => {

    const { user, setUser} = useContext( Usercontext );
    return (
        <>
            <h1>LoginPage</h1>
            <hr />
            <pre aria-label="pre">
                {JSON.stringify(user, null, 3) }
            </pre>
            <button aria-label="btnSetUser" className="btn btn-primary" onClick={ () => setUser({id: 123, name: 'juan', email: 'asjda@gmail.com'})}>

                establecer usuario
            </button>
        </>
    )
}