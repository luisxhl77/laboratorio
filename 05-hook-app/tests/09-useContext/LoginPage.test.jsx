import { fireEvent, render, screen } from "@testing-library/react"
import { Usercontext } from "../../src/09-useContext/context/userContext"
import { LoginPage } from "../../src/09-useContext/LoginPage"

describe('Pruebas de <LoginPage/>', () => {
    const user = {
        id: 123, 
        name: 'juan', 
        email: 'asjda@gmail.com'
    }

    test('bede de hacer match con el snapshot', () => {

        const { container } = render( 
            <Usercontext.Provider value={{user: null}}>
                <LoginPage/> 
            </Usercontext.Provider>
        );
        expect( container ).toMatchSnapshot();
    })

    test('bede de mostrar el complemento sin el usuario', () => {
        const { container } = render( 
            <Usercontext.Provider value={{user: null}}>
                <LoginPage/> 
            </Usercontext.Provider>
        );
        
        const preTag = screen.getByLabelText('pre');
        expect ( preTag.innerHTML ).toBe('null');
            
    })

    test('bede de llamar al setUser cuando se hace click en el boton', () => {

        const setUserMock = jest.fn()

        const { container } = render( 
            <Usercontext.Provider value={{user: null, setUser: setUserMock}}>
                <LoginPage/> 
            </Usercontext.Provider>
        );
        const btnSetUser = screen.getByLabelText('btnSetUser');
        fireEvent.click(btnSetUser);

        expect( setUserMock ).toHaveBeenCalled();
        expect( setUserMock ).toHaveBeenCalledWith( user );
    })
})