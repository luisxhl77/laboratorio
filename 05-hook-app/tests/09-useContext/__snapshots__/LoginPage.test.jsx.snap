// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <LoginPage/> bede de hacer match con el snapshot 1`] = `
<div>
  <h1>
    LoginPage
  </h1>
  <hr />
  <pre
    aria-label="pre"
  >
    null
  </pre>
  <button
    aria-label="btnSetUser"
    class="btn btn-primary"
  >
    establecer usuario
  </button>
</div>
`;
